File: task_equivalents/llm_translations/gpt/optimized/r/data_manipulation_preprocessing/02_merge_aggregate_reshape.R
library(readr)
library(dplyr)

College <- read_csv("data/csv/College.csv", show_col_types = FALSE)

school_name <- College[[1]]
College <- College |>
tibble::column_to_rownames(var = names(College)[1])

College <- College |>
mutate(Private01 = if_else(Private == "Yes", 1, 0)) |>
select(-Private)

X <- College |>
select(-Apps) |>
mutate(across(everything(), as.double))

Y <- College$Apps

set.seed(1)
train_idx <- sample.int(nrow(College), size = floor(0.5 * nrow(College)))
xtrain <- X[train_idx, , drop = FALSE]
xtest <- X[-train_idx, , drop = FALSE]
ytrain <- Y[train_idx]
ytest <- Y[-train_idx]

File: task_equivalents/llm_translations/gpt/optimized/python/data_manipulation_preprocessing/02_merge_aggregate_reshape.py
import random
import pandas as pd

college = pd.read_csv("data/csv/College.csv")

college.index = college.iloc[:, 0]
college = college.iloc[:, 1:].copy()

college["Private01"] = (college["Private"] == "Yes").astype(int)
college = college.drop(columns=["Private"])

x = college.drop(columns=["Apps"]).apply(pd.to_numeric)
y = college["Apps"]

random.seed(1)
train = random.sample(range(len(college)), len(college) // 2)
test = sorted(set(range(len(college))) - set(train))

xtrain = x.iloc[train]
xtest = x.iloc[test]
ytrain = y.iloc[train]
ytest = y.iloc[test]